flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java

```diff
@@ -173,6 +186,22 @@ public static StateBackend loadStateBackendFromConfig(
         }
     }
 
+    public static StateBackend loadWrappedStateBackendFromConfig(
+            ReadableConfig config, ClassLoader classLoader, @Nullable Logger logger)
+            throws IllegalConfigurationException, DynamicCodeLoadingException, IOException {
```

Comment:
This changes adds two new public methods, increasing the total number of methods to get a backend to four. Ideally, I think there should be just one public method.

I see that two existing methods are now marked with `@VisibleForTesting` but that's not clear enough I think.

Instead, why not
1. Make existing methods private and add some suffix like `private loadStateBackendFromConfig*Internal`
2. Add two corresponding methods (`loadStateBackendFromConfig` and `fromApplicationOrConfigOrDefault`) which would call original ones and wrap if needed
3. No changes to callers required
?

Reply:
Yes, that's a good idea. But there might be some more clean-up tests, but I like this idea better.

Reply:
This is done.